Added details of new Pipeline Setting (#4618)
harness/developer-hub: docs/platform/settings/default-settings.md

- **Enable Pipeline Execution Audit Events:** By default, the pipeline execution audit events such as Pipeline Start, Pipeline End, Stage Start, and Stage End are not displayed in Audit Trail. Enable this setting to display these events. For more information, go to [View an Audit Trail](/docs/platform/governance/audit-trail/#step-view-an-audit-trail).

> - **Run RBAC Validation before executing Inline Pipelines:** By default, before running any Pipeline, Harness checks whether the running User has access to all of the Environments and other resources that the Pipeline accesses. This check is run only for Inline Pipelines, and not for Pipelines which are stored in Git or other repositories. Turn off this setting if the check is not required. Turning it off may cause Pipelines to fail partway through execution (since later stages could have permission issues which haven't been checked beforehand).

Reviewer: ```suggestion
- **Run RBAC Validation before executing Inline Pipelines:** By default, before running any pipeline, Harness checks whether the user has access to all of the environments and other resources that the pipeline accesses. This check is run only for inline pipelines, not those stored in Git or other repositories. Turn off this setting if the check isn't required. Turning it off may cause pipelines to fail partway through execution (since later stages could have permission issues that haven't been checked beforehand).
```
